frontend/src/containers/Dashboard/actions.js

```diff
@@ -1,16 +1,60 @@
-export const getHouseIssues = () => {
-  return new Promise( (resolve, reject) => {
-    resolve(
-      [
-        {
-          _id: 1,
-          house_image: "//via.placeholder.com/200x200",
-          issues: [
-            {_id: 1, status: "open", title: "test issue", description: "This is a test issue"},
-            {_id: 2, status: "resolved", title: "kitchen tap leaking", description: "can't turn the kitchen tap off!"}
-          ]
-        }
-      ]
-    );
-  });
-};
+export const HOUSE_ISSUES_HAS_ERRORED = "HOUSE_ISSUES_HAS_ERRORED";
+export const HOUSE_ISSUES_IS_LOADING = "HOUSE_ISSUES_IS_LOADING";
+export const HOUSE_ISSUES_FETCH_DATA_SUCCESS = "HOUSE_ISSUES_FETCH_DATA_SUCCESS";
+
+export function houseIssuesHasErrored( bool) {
+  return {
+    type: HOUSE_ISSUES_HAS_ERRORED,
+    hasErrored: bool
+  };
+}
+
+export function houseIssuesIsLoading( bool) {
+  return {
+    type: HOUSE_ISSUES_IS_LOADING,
+    isLoading: bool
+  };
+}
+
+export function houseIssuesFetchDataSuccess( houseIssues) {
+  return {
+    type: HOUSE_ISSUES_FETCH_DATA_SUCCESS,
+    houseIssues
+  };
+}
+
+export function houseIssuesFetchData( url) {
+  return (dispatch) => {
+    dispatch( houseIssuesHasErrored( false));
+    dispatch( houseIssuesIsLoading( true));
+    fetch( url)
+    .then( response => {
+      if( !response.ok){
+        throw Error( response.statusText);
+      }
+      dispatch( houseIssuesIsLoading( false));
+      return response;
+    })
+    .then( response => response.json())
+    .then( house_issues => dispatch( houseIssuesFetchDataSuccess( house_issues.houses)))
```

Comment:
This is my mind should be two actions. I.e. one for success `true or false` and one for `addHouseIssues` It doesn't make sense to store the houseIssues using fetchDataSuccess. I think? Second thoughts you could use has Errored to see if is successful or not. But then rename `houseIssuesFetchDataSuccess` to something else. It's slightly misleading.

Reply:
First time implementing redux, so pretty much copied from an example that worked and that I could understand. I'll have to come back to this one and have a thnk after doing the simple changes.

Reply:
@Mortuie you mean instead of dispatching another event, just return action object? e.g. 
```
  return {
    type: HOUSE_ISSUES_FETCH_DATA_SUCCESS,
    houseIssues: house_issues.houses
  };
```

Reply:
@Mortuie well that didn't work :)   Could it be because it's an action creator, not an action? Can you suggest what the code should look like?